pkg/apis/fleet/v1alpha1/types.go

```diff
@@ -150,13 +149,10 @@ type ThanosConfig struct {
 	//
 	// +optional
 	Chart *ChartConfig `json:"chart,omitempty"`
-	// Mode defines the mode of the Thanos.
-	// The default is sidecar.
-	// +kubebuilder:validation:Enum=sidecar;reciever
-	// +optional
-	Mode string `json:"mode,omitempty"`
-	// ObjectStoreConfig defines the configuration for the object store.
-	ObjectStore *ThanosObjectStoreConfig `json:"objectStore,omitempty"`
```

Comment:
Why remove, donot we need to specify where to store the data?

Reply:
Most of thanos user will create a storage secret, so change this to a secret referenece.